### ISSUE TITLE
MonkeyRunner.waitForConnection() wakes device from sleep

### ISSUE THREAD

**Issue by clapbr** [nlitsme/PythonMonkey#3]
Not sure if this is intended behavior but every time I reconnect to the device using IE `MonkeyRunner.waitForConnection(10)` it turns the screen on from sleep (I want it to connect "silently" to my TV).

I tried following the code path but I couldn't figure which part does that, any idea?

**Comment by nlitsme** [comment]
In my code, i ping the monkey connection by executing the 'wake' command. Changing that to `mon.getvar("monkey.version")`  did not solve this, looking deeper, i found that in google's code, in the function [`AdbChimpDevice.createManager`](https://android.googlesource.com/platform/tools/swt/+/master/chimpchat/src/main/java/com/android/chimpchat/adb/AdbChimpDevice.java#195) the `wake` command is issued too.

**Comment by clapbr** [comment]
> In my code, i ping the monkey connection by executing the 'wake' command. Changing that to `mon.getvar("monkey.version")` did not solve this, looking deeper, i found that in google's code, in the function [`AdbChimpDevice.createManager`](https://android.googlesource.com/platform/tools/swt/+/master/chimpchat/src/main/java/com/android/chimpchat/adb/AdbChimpDevice.java#195) the `wake` command is issued too.

Thanks for the quick reply, I've just tried the official jython monkeyrunner and it also turns the device on confirming your findings, seems we're stuck with it since the code you linked lives in Android OS itself. I'll workaround it on my app by checking states via adb before calling waitForConnection().

edit: btw my app is a simple rest API backend for remote controlling my TV, here's the rough draft https://gist.github.com/clapbr/f525c15c2795bdd87041022c5f759390 - I consume it from Home Assistant atm, but it's so simple it can be easily used by anything. It's my first coding project ever and security was totally negleted for now, but it's working very well. I used "adb input" before but latency is too high (~2s)
